src/blog/2022/04/flowforge-04-released.md

```diff
@@ -0,0 +1,97 @@
+---
+title: FlowForge 0.4 released
+subtitle: Getting ready for Node-RED 3.0
+description: Getting ready for Node-RED 3.0
+date: 2022-04-14 12:00:00.0
+authors: ["sam-machin"]
+---
+
+This release of the FlowForge add a seemingly small but significant new feature.
+
+<!--more-->
+
+With (Node-RED 3.0 fast approaching)[https://nodered.org/about/releases/] we've been making sure we are ready to support this.
+
+### Upgrading Node-RED
+
+The goal of FlowForge is to be the best way to run Node-RED at scale, whether thats many users or many instances. Node-RED is a constantly developing as a platform and therefore part of running Node-RED is also upgrading the version you are running.
+
+With the 0.4 release today we've made that super simple in FlowForge. Last month we introduced the concept of [Project Stacks](https://flowforge.com/docs/user/concepts/#project-stack), and one of the key elements of a Stack was the version of Node-RED in use. Initially this may have seemed fairly basic, when you create a new project you usually want to use the latest version of Node-RED. However what happens when a new version is released and you have an existing project?
+Now you can change the stack that a project is running on, which in turn will change the version of Node-RED. This is a simple process from the project settings, it only requires a short period of downtime while the project restarts on the new stack, typically around 10-15 seconds.
+
+Our driver to get this feature into the 0.4 release is the approaching release of Node-RED 3.0, now we know that we can be ready to offer our users Node-RED 3.0 as soon as it is released.
+
+We will also be making available the Beta's of Node-RED 3.0 within FlowForge Cloud,  this becomes a great way to test out the new features without having to touch your own environments.
```

Comment:
Who hosts our code and runs it, is an implementation detail. There's no such thing as "FlowForge Cloud", it's just FlowForge.

Also; we should say when it's available. And if possible, let's create that stack today. That changes the messaging to:
```suggestion
The beta releases of Node-RED 3.0 are available today. This is a great way to test new features without having to touch your own environments.
```

Reply:
We can't create the stack yet, because there hasn't been a NR 3.0.0 beta release yet

Reply:
we're a good 2 weeks away from Node-RED beta 1, so this would be way to early, we can do another new post when we make the stack available.

Reply:
Almost want a `node-red-master` stack now, though the instability of the underlying containers is probably not a good idea. Will mark resolved.

Reply:
Yeah I did think about Node-RED Nightly Build, but I'm not quite sure we're there with the level of automation needed to build a new container just yet. Its an interesting idea though